### ISSUE TITLE
Grant Project Progress Reporting: Enable did:elem in BRI-1

### ISSUE THREAD

**Issue by Therecanbeonlyone1969** [eea-oasis/baseline-grants#46]
Starting with a September progress report during the 1st week of October, Spruce is to provide a monthly progress update no later than the 1st Friday of the month by leaving report comments with relevant documents on this issue.

The progress report must answer the following questions:

1. What was achieved during the previous month by achieved deliverable
2. What is intended to be achieved during the upcoming month
3. Are there any blockers / issues that would impact the delivery timeline of 12/31/2021, and if yes, what are they, and what are the mitigation measures taken to mitigate the risk.

cc @bumblefudge

**Comment by bumblefudge** [comment]
First Friday is tomorrow!  We had three months budgeted for this topic, broken down as follows:
1. October - NATS and Provide modifications, requirements for Sidetree
2. November - Integration and Testing 
3. December - Documentation and discussion

What is intended to be achieved during the upcoming month

We've been looking at [JWT-mode NATS configuration](https://docs.nats.io/nats-server/configuration/securing_nats/jwt) and the [patch mechanics](https://identity.foundation/sidetree/spec/#standard-patch-actions) for adding an additional Ed25519 key to Sidetree DID Documents and will spend next week doing some more research and planning before we start implementing the necessary changes in BR-1.

Are there any blockers / issues that would impact the delivery timeline of 12/31/2021, and if yes, what are they, and what are the mitigation measures taken to mitigate the risk.

Not blockers exactly but I will reach out now for a meeting with Kyle to validate our plan of attack midmonth. We've started on the changes to existing components to frontload/parallelize the work, but if Transmute were delayed substantially in their reference implementation, we are already using an older version of Sidetree to validate our assumptions and in theory we could even start documentation and the hello-world video before swapping it out for the newer version.

**Comment by bumblefudge** [comment]
Quick updates:
* NATS and Provide integration went well! draft PR open on [provide/ident](https://github.com/provideplatform/ident/pull/10) for the curious
* Met with Kyle to discuss PR above, worked out a kink in the permissioning model
   - we also talked about some possible out-of-scope work, namely, calling didkit as a library from behind the API to expose a "/did/resolve" endpoint that returns a did document for a supported (configurable) subset of did methods. this is beyond scope for our integration, but might be useful for other use-cases or even as a fallback.
* We're ready to start working with a Orie's [working draft](https://github.com/decentralized-identity/element) this month!

**Comment by bumblefudge** [comment]
Quick updates:
- Experienced some internal delays in November and are still transferring over the general-purpose DID work to did:elem-specific DID work and integrating the Sidetree node to the BR-1 fork
- still on track to finish this month

**Comment by bumblefudge** [comment]
Our draft mono-repo is up, for people who want to take a look or do an initial review:
[provide mono-repo](https://github.com/theosirian/provide-did](https://github.com/theosirian/provide-did)
Also included is our release checklist of remaining [issues](https://github.com/theosirian/provide-did/issues/1)
